src/Orleans/EventSourcing/JournaledGrainState.cs

```diff
@@ -28,36 +28,58 @@ MIT License
 
 namespace Orleans.EventSourcing
 {
-    public abstract class JournaledGrainState : GrainState
+    public abstract class JournaledGrainState<TGrainState> : GrainState
+        where TGrainState : JournaledGrainState<TGrainState>
     {
         private List<StateEvent> events = new List<StateEvent>();
-        protected JournaledGrainState(Type type )
-            : base(type.FullName)
```

Comment:
If this is a 'GrainState' shouldn't it support the same construction pattern as grain state?  I.E. JournaledGrainState() and JournaledGrainState(string name)